src/rgw/rgw_sync.cc

```diff
@@ -447,8 +447,13 @@ class RGWReadMDLogEntriesCR : public RGWSimpleCoroutine {
       entries(_entries), truncated(_truncated) {}
 
   ~RGWReadMDLogEntriesCR() override {
+    request_cleanup();
```

Comment:
I think the extra call here isn't necessary, as request_cleanup() is called in ~RGWSimpleCoroutine() if it's never been called.  So just implementing request_cleanup() should be sufficient.  (Note, because it's all conditional on req, it doesn't make a difference here, but it could cause a problem if more is added later.)